File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/ListViews_ViewModel.cpp

#include "ListViews_ViewModel.h"
#include <vector>
#include <memory>
#include "ListViews_ViewModelMyListOfLabelRow.h"
#include "ListViews_ViewModelMyListOfImagesRow.h"
#include "ListViews_ViewModelMyCheckboxesRow.h"
#include "ListViews_ViewModelMyStringRowHandlesRow.h"
#include "ListViews_ViewModelMyListWithMultiRowSelectionRow.h"
#include "ListViews_ViewModelMyButtonsRow.h"
#include <string>

namespace widgetassertions
{
  bool ListViews_ViewModel::getIsMyListOfLabelListVisible( ) 
  {
    return this->isMyListOfLabelListVisible;
  }
  bool ListViews_ViewModel::getIsMyListOfLabelListEnabled( ) 
  {
    return this->isMyListOfLabelListEnabled;
  }
  std::vector<std::shared_ptr<ListViews_ViewModelMyListOfLabelRow>> ListViews_ViewModel::getMyListOfLabelListRows( ) 
  {
    return this->MyListOfLabelListRows;
  }
  int ListViews_ViewModel::getMyListOfLabelListSelectedRow( ) 
  {
    return this->MyListOfLabelListSelectedRow;
  }
  bool ListViews_ViewModel::getIsMyListOfImagesListVisible( ) 
  {
    return this->isMyListOfImagesListVisible;
  }
  bool ListViews_ViewModel::getIsMyListOfImagesListEnabled( ) 
  {
    return this->isMyListOfImagesListEnabled;
  }
  std::vector<std::shared_ptr<ListViews_ViewModelMyListOfImagesRow>> ListViews_ViewModel::getMyListOfImagesListRows( ) 
  {
    return this->MyListOfImagesListRows;
  }
  int ListViews_ViewModel::getMyListOfImagesListSelectedRow( ) 
  {
    return this->MyListOfImagesListSelectedRow;
  }
  std::vector<std::shared_ptr<ListViews_ViewModelMyCheckboxesRow>> ListViews_ViewModel::getMyCheckboxesListRows( ) 
  {
    return this->MyCheckboxesListRows;
  }
  int ListViews_ViewModel::getMyCheckboxesListSelectedRow( ) 
  {
    return this->MyCheckboxesListSelectedRow;
  }
  bool ListViews_ViewModel::getIsMyCheckboxesListVisible( ) 
  {
    return this->isMyCheckboxesListVisible;
  }
  bool ListViews_ViewModel::getIsMyCheckboxesListEnabled( ) 
  {
    return this->isMyCheckboxesListEnabled;
  }
  std::vector<std::shared_ptr<ListViews_ViewModelMyStringRowHandlesRow>> ListViews_ViewModel::getMyStringRowHandlesListRows( ) 
  {
    return this->MyStringRowHandlesListRows;
  }
  std::string ListViews_ViewModel::getMyStringRowHandlesListSelectedRow( ) 
  {
    return this->MyStringRowHandlesListSelectedRow;
  }
  std::vector<std::shared_ptr<ListViews_ViewModelMyListWithMultiRowSelectionRow>> ListViews_ViewModel::getMyListWithMultiRowSelectionListRows( ) 
  {
    return this->MyListWithMultiRowSelectionListRows;
  }
  std::vector<int> ListViews_ViewModel::getMyListWithMultiRowSelectionListSelectedRows( ) 
  {
    return this->MyListWithMultiRowSelectionListSelectedRows;
  }
  std::vector<std::shared_ptr<ListViews_ViewModelMyButtonsRow>> ListViews_ViewModel::getMyButtonsListRows( ) 
  {
    return this->MyButtonsListRows;
  }
  void ListViews_ViewModel::buttonsClicked(int rowIndex ) 
  {
    
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/ListViews_ViewModel.cs
/// <filename>
///     ListViews_ViewModel.cs
/// </filename>
using System.Collections.Generic;
using widgetassertions;

namespace widgetassertions
{
    public class ListViews_ViewModel
    {
        private bool isMyListOfLabelListVisible;

        public virtual bool getIsMyListOfLabelListVisible()
        {
            return this.isMyListOfLabelListVisible;
        }

        private bool isMyListOfLabelListEnabled;

        public virtual bool getIsMyListOfLabelListEnabled()
        {
            return this.isMyListOfLabelListEnabled;
        }

        private System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListOfLabelRow> MyListOfLabelListRows = new System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListOfLabelRow>();

        public virtual System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListOfLabelRow> getMyListOfLabelListRows()
        {
            return this.MyListOfLabelListRows;
        }

        private int MyListOfLabelListSelectedRow;

        public virtual int getMyListOfLabelListSelectedRow()
        {
            return this.MyListOfLabelListSelectedRow;
        }

        private bool isMyListOfImagesListVisible;

        public virtual bool getIsMyListOfImagesListVisible()
        {
            return this.isMyListOfImagesListVisible;
        }

        private bool isMyListOfImagesListEnabled;

        public virtual bool getIsMyListOfImagesListEnabled()
        {
            return this.isMyListOfImagesListEnabled;
        }

        private System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListOfImagesRow> MyListOfImagesListRows = new System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListOfImagesRow>();

        public virtual System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListOfImagesRow> getMyListOfImagesListRows()
        {
            return this.MyListOfImagesListRows;
        }

        private int MyListOfImagesListSelectedRow;

        public virtual int getMyListOfImagesListSelectedRow()
        {
            return this.MyListOfImagesListSelectedRow;
        }

        private System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyCheckboxesRow> MyCheckboxesListRows = new System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyCheckboxesRow>();

        public virtual System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyCheckboxesRow> getMyCheckboxesListRows()
        {
            return this.MyCheckboxesListRows;
        }

        private int MyCheckboxesListSelectedRow;

        public virtual int getMyCheckboxesListSelectedRow()
        {
            return this.MyCheckboxesListSelectedRow;
        }

        private bool isMyCheckboxesListVisible;

        public virtual bool getIsMyCheckboxesListVisible()
        {
            return this.isMyCheckboxesListVisible;
        }

        private bool isMyCheckboxesListEnabled;

        public virtual bool getIsMyCheckboxesListEnabled()
        {
            return this.isMyCheckboxesListEnabled;
        }

        private System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyStringRowHandlesRow> MyStringRowHandlesListRows = new System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyStringRowHandlesRow>();

        public virtual System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyStringRowHandlesRow> getMyStringRowHandlesListRows()
        {
            return this.MyStringRowHandlesListRows;
        }

        private string MyStringRowHandlesListSelectedRow;

        public virtual string getMyStringRowHandlesListSelectedRow()
        {
            return this.MyStringRowHandlesListSelectedRow;
        }

        private System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListWithMultiRowSelectionRow> MyListWithMultiRowSelectionListRows = new System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListWithMultiRowSelectionRow>();

        public virtual System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyListWithMultiRowSelectionRow> getMyListWithMultiRowSelectionListRows()
        {
            return this.MyListWithMultiRowSelectionListRows;
        }

        private System.Collections.Generic.List<int> MyListWithMultiRowSelectionListSelectedRows = new System.Collections.Generic.List<int>();

        public virtual System.Collections.Generic.List<int> getMyListWithMultiRowSelectionListSelectedRows()
        {
            return this.MyListWithMultiRowSelectionListSelectedRows;
        }

        private System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyButtonsRow> MyButtonsListRows = new System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyButtonsRow>();

        public virtual System.Collections.Generic.List<widgetassertions.ListViews_ViewModelMyButtonsRow> getMyButtonsListRows()
        {
            return this.MyButtonsListRows;
        }

        public virtual void buttonsButtonClicked(int rowIndex)
        {
        }
    }
}
